tests/unit/utils/test_src_utils.py

```diff
@@ -51,3 +56,84 @@ def test_is_empty_str(input_data, expected_output):
 )
 def test_is_non_empty_str(input_data, expected_output):
     assert utils.is_non_empty_str(input_data) == expected_output
+
+
+@pytest.mark.parametrize(
+    "test_description, token, expected_result, expected_exception",
+    [
+        (
+            "valid token with sub",
+            jwt.encode({"sub": "user123"}, "secret", algorithm="HS256"),
+            {"sub": "user123"},
+            None,
+        ),
+        (
+            "invalid token",
+            "invalid.token.here",
+            None,
+            ValueError,
+        ),
+        (
+            "empty token",
+            "",
+            None,
+            ValueError,
+        ),
+    ],
+)
+def test_parse_k8s_token(test_description, token, expected_result, expected_exception):
+    if expected_exception:
+        with pytest.raises(expected_exception, match="Failed to parse k8s token"):
+            parse_k8s_token(token)
+    else:
+        decoded_token = parse_k8s_token(token)
+        assert decoded_token == expected_result
+
+
+@pytest.mark.parametrize(
+    "test_description, token_payload, expected_result, expected_exception",
+    [
+        (
+            "valid token with sub",
+            {"sub": "user123"},
+            "user123",
+            None,
+        ),
+        (
+            "valid token with email",
+            {"email": "user@example.com"},
+            "user@example.com",
+            None,
+        ),
+        (
+            "valid token with service account name",
+            {"kubernetes.io/serviceaccount/service-account.name": "service-account"},
+            "service-account",
+            None,
+        ),
```

Comment:
nitpick: you can re-use the constants SUB, EMAIL, SERVICE_ACCOUNT_NAME here to if you define them in the main code.

Reply:
done